server.js

```diff
@@ -1,55 +1,52 @@
 // load .env data into process.env
-require("dotenv").config();
+require('dotenv').config();
 
 // Web server config
```

Comment:
I have noticed that being more explicit about the different parts of the server code help the students to understand how things are connected between them, I always like to separate it as following:

- **Requirements:** all the require statements starting from modules and then local helper files
- **Setup and middlewares:** creating instances and adding app.use middlewares
- **Routes/endpoints:** all the possible ways to interact and persist data with your server/API
- **Listener:** opening the doors of your server to accept incoming requests

A reference of a code that I created for a lecture following this pattern can be found in the next link: https://github.com/pedroagont/webflex-lectures-21mar/tree/main/m05w12

It also includes a reference on how to apply the MVC Architecture by isolating both the controllers and models to accurately query the database, I understand this is currently being handled by the queries directory but I just mention this in case it helps. Also is important to say that this code I just shared includes a simple but full CRUD implementation so the create, getAll, getById, update, delete and remove routes+controller+model for any resources (in this case called fruit) so it can also be used as a template

Hope it helps team! 🚀👨🏻‍🏫🌴

Reply:
Thanks for this! I moved like-with-like in the server file for the most part. I've left the requires of the routers and their app.use's together for teaching purposes.

I appreciate the MVC suggestion! But the midterm is designed to reinforce what the students have already seen. At this point in the program (midterm), students haven't been exposed to an MVC yet. We also only spend one day on object-oriented JS in the program. The students will get MVC and OOP experience in the Rails portion of the program. 😺